torch/csrc/autograd/custom_function.cpp

```diff
@@ -339,6 +349,7 @@ optional_variable_list _process_backward_mode_ad(
         impl::rebase_history(var, {cdata, output_nr});
       }
     } else if (is_input) {
+      TORCH_CHECK(!is_saved_and_setup_context, error_msg_input_returned_as_is)
```

Comment:
Why don't we need a !is_modified check here?

Reply:
`if (is_modified) { ... } else if (is_input) { <we are here!> }`

Reply:
Oh lol the nesting got to me. Thanks for pointing that out!